ansible/playbooks/roles/repository/files/download-requirements/src/downloader.py

```diff
@@ -54,18 +54,22 @@ def __is_checksum_valid(self, requirement: str, requirement_file: Path) -> bool:
 
         return True
 
-    def download(self, requirement: str, requirement_file: Path, sub_key: str = None):
+    def download(self, requirement: str, requirement_file: Path, sub_key: str = None, additional_args: dict = None):
         """
         Download `requirement` as `requirement_file` and compare checksums.
 
         :param requirement: an entry from the requirements corresponding to the downloaded file
         :param requirement_file: existing requirement file
         :param sub_key: optional key for the `requirement` such as `url`
+        :param additional_args: optional arguments passed to `download_func`
         :raises:
             :class:`ChecksumMismatch`: can be raised on failed checksum
         """
         req = self.__requirements[requirement][sub_key] if sub_key else requirement
 
+        if additional_args:
+            self.__download_args.update(additional_args)
```

Comment:
I don't think that it's safe to update self.__download_args with those additional_args. Since self.__download_args is used between each download() call. It would be best to merge those args on the fly as a copy.